docs/tutorials/text_sentiment_ngrams.ipynb

```diff
@@ -0,0 +1,465 @@
+    "        Acc: 97.2%(train)       |       96.1%(valid)\n",
+    "        \n",
+    "Epoch: 8  | time in 0 minutes, 5 seconds\n",
+    "\n",
+    "        Loss: 0.0013(train)     |       0.0014(valid)\n",
+    "        Acc: 97.9%(train)       |       97.7%(valid\n",
+    "        \n",
+    "Epoch: 9  | time in 0 minutes, 5 seconds\n",
+    "\n",
+    "        Loss: 0.0011(train)     |       0.0012(valid)\n",
+    "        Acc: 98.3%(train)       |       98.1%(valid)\n",
+    "        \n",
+    "Epoch: 10  | time in 0 minutes, 5 seconds\n",
+    "\n",
+    "        Loss: 0.0010(train)     |       0.0010(valid)\n",
+    "        Acc: 98.6%(train)       |       98.7%(valid\n",
+    "        \n",
+    "Epoch: 11  | time in 0 minutes, 5 seconds\n",
+    "\n",
+    "        Loss: 0.0008(train)     |       0.0008(valid)\n",
+    "        Acc: 98.9%(train)       |       99.0%(valid)\n",
+    "        \n",
+    "Epoch: 12  | time in 0 minutes, 5 seconds\n",
+    "\n",
+    "        Loss: 0.0007(train)     |       0.0008(valid)\n",
+    "        Acc: 99.1%(train)       |       99.1%(valid)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Evaluate the model with test dataset"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print('Checking the results of test dataset...')\n",
+    "with torch.no_grad():\n",
+    "    test_loss, test_acc = train_evaluate_func(model, test_dataset, loss_func,\n",
+    "                                              device=device, status='test')\n",
+    "print(f'\\tLoss: {test_loss:.4f}(test)\\t|\\tAcc: {test_acc * 100:.1f}%(test)')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Checking the results of test dataset...\n",
+    "\n",
+    "        Loss: 0.005(test)       |       Acc: 91.0%(test\n",
+    "        \n",
+    "The results are consistent with the reference paper [Bag of Tricks for Efficient Text Classification](https://arxiv.org/abs/1607.01759)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Test on a random news\n",
+    "\n",
+    "Use the best model so far and test a golf news."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import re\n",
+    "from torchtext.data.utils import ngrams_iterator\n",
+    "from torchtext.data.utils import get_tokenizer\n",
+    "\n",
+    "ag_news_label = {1 : \"World\",\n",
+    "                 2 : \"Sports\",\n",
+    "                 3 : \"Business\",\n",
+    "                 4 : \"Sci/Tec\"}\n",
+    "\n",
+    "vocab = train_dataset._vocab\n",
+    "\n",
+    "def predict_label(text_string, ngrams):\n",
+    "    model.eval()\n",
+    "    tokenizer = get_tokenizer(\"basic_english\")\n",
+    "    ngrams_tokens = ngrams_iterator(tokenizer(text_string), ngrams)\n",
```

Comment:
you call ngrams_iterator twice

you also don't need to create offsets if you just pass in a single line.

Look at the predict.py example to see how it's done there.